docs/tutorials/sinks/1_basic_sink.md

```diff
@@ -0,0 +1,340 @@
+        toml::from_str("").unwrap()
+    }
+}
+```
+
+This is used by the `vector generate` command to generate a default
+configuration for the sink.
+
+# SinkConfig
+
+We need to implement the `SinkConfig` trait. This is used by Vector to generate
+the main Sink from the configuration.
+
+```rust
+#[async_trait::async_trait]
+impl SinkConfig for BasicConfig {
+    async fn build(&self, _cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
+        let healthcheck = Box::pin(async move { Ok(()) });
+        let sink = VectorSink::from_event_streamsink(BasicSink);
+
+        Ok((sink, healthcheck))
+    }
+
+    fn input(&self) -> Input {
+        Input::log()
+    }
+
+    fn acknowledgements(&self) -> &AcknowledgementsConfig {
+        &self.acknowledgements
+    }
+}
+```
+
+## The `build` function
+
+Of particular importance is the `build` function. This is an async function that
+builds two components of the sink.
+
+First, the healthcheck is an async block that can be used to check the health
+of the service we are connecting to. In this very simple case we are just
+outputting to the console which we assume will work, so the healthcheck returns
+`Ok(())` indicating our target is healthy.
+
+The actual work for this sink is done in `BasicSink` (to be implemented
+shortly). The `build` function converts this into a `VectorSink` via
+`VectorSink::from_event_streamsink` and returns it.
+
+
+## BasicSink
+
+Lets implement `BasicSink`.
+
+```rust
+struct BasicSink;
+```
+
+Our sink is so basic it has no properties to determine it's behaviour.
+
+For it to work with Vector it must implement the `StreamSink` trait:
+
+```rust
+#[async_trait::async_trait]
+impl StreamSink<Event> for BasicSink {
+    async fn run(
+        self: Box<Self>,
+        input: futures_util::stream::BoxStream<'_, Event>,
+    ) -> Result<(), ()> {
+        self.run_inner(input).await
+    }
+}
+```
+
+`StreamSink` is an async trait with a single async function: `run`. The main
+parameter to this function, `input` is a stream of the events that are being
+sent to this sink. We pull from this stream to send the events on to our
+destination.
+
+In order to handle lifetime issues that arise from using  [`async_trait`]
+(https://docs.rs/async-trait/latest/async_trait/), this function simply calls
+another method `run_inner` that is implemented directly on `BasicSink`.
+
+Let's look an `run_inner`:
+
+```rust
+impl BasicSink {
+    async fn run_inner(self: Box<Self>, mut input: BoxStream<'_, Event>) -> Result<(), ()> {
+        while let Some(event) = input.next().await {
+            println!("{:?}", event);
+        }
+
+        Ok(())
+    }
+}
+```
+
+Our sink simply pulls each event from the input stream and prints the debug
+representation of the object.
+
+# Importing to Vector
+
+Import this module into Vector. In `src/sinks/mod.rs` add the line:
+
+
+```diff
+ #[cfg(feature = "sinks-azure_monitor_logs")]
+ pub mod azure_monitor_logs;
+
++pub mod basic;
```

Comment:
I might include some notes about only building Vector with your new component `--no-default-features --features ...` to aid contributor development cycles.

Reply:
I was thinking to avoid talking about features here because I thought it would be just extraneous details that could confuse the issue, but this is a very valid idea! 

I should add a section on `vdev` too..